moped-editor/src/views/projects/projectView/ProjectComponents/EditAttributesModal.js

```diff
@@ -28,23 +35,16 @@ const EditAttributesModal = ({
   editDispatch,
   componentToEdit,
   refetchProjectComponents,
-  setClickedComponent,
   mapRef,
 }) => {
   const classes = useStyles();
 
   const [updateComponentAttributes] = useMutation(UPDATE_COMPONENT_ATTRIBUTES);
   const [updateSignalComponent] = useMutation(UPDATE_SIGNAL_COMPONENT);
 
-  const onComponentSaveSuccess = (updatedClickedComponentState) => {
-    // Update component list item and clicked component state to keep UI up to date
+  const onSaveSuccess = () => {
     refetchProjectComponents().then(() => {
       onClose();
-      // Update clickedComponent with the attributes that were just edited
-      setClickedComponent((prevComponent) => ({
-        ...prevComponent,
-        ...updatedClickedComponentState,
-      }));
```

Comment:
This set state call is not used anymore and is what caused the momentary flash of old values. We now:
1. Submit the edit mutation
2. Refetch the data about the updated component
3. Close the modal

This removes the effort of needing to make the form data into the shape returned by Hasura which is stored in `clickedComponent` state. Now, the updated data returns and then populates `clickedComponent` state instead of being set by the code here.